mint-client/src/bin/clientd/main.rs

```diff
@@ -33,43 +25,153 @@ struct Opts {
+    amount : Amount,
+}
+#[derive(Serialize)]
+struct CoinGrouped{
+    amount : usize,
+    tier : u64
+}
+#[derive(Serialize)]
+struct SpendResponse {
+    token : String,
+}
+
+#[derive(Serialize)]
+struct ReissueResponse {
+    id : String,
+    fetched : Vec<TransactionId>,
+}
+
+#[derive(Deserialize,Serialize, Debug)]
+struct ReqBody<T> {
+    //I dont like this
+    value : T
+}
+
+impl InfoResponse {
+    fn new(coins: Coins<SpendableCoin>) -> Self {
+        let info_total =  CoinTotal { coin_count: coins.coin_count(), amount: coins.amount() };
+        let info_coins : Vec<CoinGrouped> = coins.coins.iter()
+            .map(|(tier, c)| CoinGrouped { amount : c.len(), tier : tier.milli_sat})
+            .collect();
+        InfoResponse { total : info_total, coins : info_coins }
+    }
+}
+
+impl ReissueResponse {
+    async fn new(mint_client :&MintClient, coins:Coins<SpendableCoin>) -> Self {
+        let mut rng = rand::rngs::OsRng::new().unwrap();
+        let id = mint_client.reissue(coins, &mut rng).await.unwrap();
+
+        let  id = id.to_hex();
+        let  fetched : Vec<TransactionId>= mint_client.fetch_all_coins().await.unwrap();
+        let res = ReissueResponse {
+            id,
+            fetched,
+        };
+        res
+    }
+}
 #[tokio::main]
 async fn main() -> tide::Result<()>{
-    //Instanciate CLient
+    //collect trace to print info
+    tracing_subscriber::fmt()
+        .with_env_filter(
+            EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("info")),
+        )
+        .init();
+
+
+    //Instantiate Client
     let opts: Opts = StructOpt::from_args();
     let cfg_path = opts.workdir.join("client.json");
     let db_path = opts.workdir.join("client.db");
-    //why use Config in ln-gateway
     let cfg: ClientConfig = load_from_file(&cfg_path);
     let db = sled::open(&db_path)
-        .unwrap()
+        .unwrap() //handle error ?
         .open_tree("mint-client")
-        .unwrap();
-    //why did ln-gateway use client.cfg ? (not in config.json)
+        .unwrap(); //handle error ?
     let client = MintClient::new(cfg, Arc::new(db), Default::default()); //Why Arc ?
 
-    //Spawn server
+
     let state = State {
         mint_client: Arc::new(client),
     };
-
     let mut app = tide::with_state(state);
-    /*
-        As far as I understood use with_state is needed to get the client into scope or to share it respectively
-        Arc is used because we can use reference-counted smart pointers (multi-threaded)
-        (thx https://www.sitepoint.com/rust-global-variables/)
-     */
-    app.at("/fetch").post(fetch);
+
+    //need move because async block could outlive the post function and needs to take ownership of req ?
+    app.at("/info").post(|req : tide::Request<State>| async move{
+        let State {
+            ref mint_client,
+        } = req.state();
+        let res = json!(InfoResponse::new(mint_client.coins()));
+        Body::from_json(&res)
```

Comment:
`from_json` is actually implemented such that any serializable object can be passed, so no need for turning it into JSON beforehand.

```rust
pub fn from_json(json: &impl Serialize) -> crate::Result<Self> { …
```